subprojects/core/src/main/java/org/gradle/internal/service/scopes/ExecutionGlobalServices.java

```diff
@@ -179,6 +179,7 @@ TypeAnnotationMetadataStore createAnnotationMetadataStore(CrossBuildInMemoryCach
             IGNORED_METHOD_ANNOTATIONS,
             method -> method.isAnnotationPresent(Generated.class),
             cacheFactory);
```

Comment:
💄 To make it more clear what is using a deprecated member I would extract some `deprecatedHasConventionClass` variable and annotate just that statement with `@SuppressWarnings` in the same way as `deprecatedAbstractTask` above.